src/main/java/org/opensearch/ad/rest/handler/IndexAnomalyDetectorActionHandler.java

```diff
@@ -164,22 +166,41 @@ public IndexAnomalyDetectorActionHandler(
         this.adTaskManager = adTaskManager;
     }
 
+    public void start() {
+        String resultIndex = anomalyDetector.getResultIndex();
+        try {
+            // use default detector result index which is system index
+            if (resultIndex == null) {
+                createOrUpdateDetector();
+                return;
+            }
+
+            // user custom result index
+            anomalyDetectionIndices.initCustomResultIndexAndExecute(resultIndex, () -> createOrUpdateDetector(), listener);
+        } catch (Exception e) {
+            logger.error("Failed to create index " + resultIndex, e);
```

Comment:
Maybe tune this error message? Couldn't this encapsulate more than just a failure to create the index? For example, in `createOrUpdateDetector()`, it calls `prepareAnomalyDetectorIndexing()`,  which does a handful of other operations (validation, creation, etc.).

Reply:
Actually we don't need to catch exception here as exception already caught by `initCustomResultIndexAndExecute` and `createOrUpdateDetector`. Will remove this try catch to make code clean.